lasy/optical_elements/optical_element.py

```diff
@@ -0,0 +1,41 @@
+import numpy as np
+
+
+class OpticalElement(object):
+    """
+    Base class to model thin optical elements.
+
+    Any optical element should inherit from this class, and define its own
+    `amplitude_multiplier` method, using the same signature as the method below.
+    """
+
+    def __init__(self):
+        pass
+
+    def amplitude_multiplier(self, x, y, omega):
+        r"""
+        Return the amplitude multiplier :math:`T`.
+
+        This number multiplies the complex amplitude of the laser
+        just before this thin element, in order to obtain the complex
+        amplitude output laser just after this thin element:
+
+        .. math::
+
+            \tilde{\mathcal{E}}_{out}(x, y, \omega) = T(x, y, \omega)\tilde{\mathcal{E}}_{in}(x, y, \omega)
+
+        Parameters
+        ----------
+        x, y, omega: ndarrays of floats
+            Define points on which to evaluate the multiplier.
+            These arrays need to all have the same shape.
```

Comment:
Should we consider 1D arrays?

Reply:
That's a good question.
I think it is easier for people to add new optical elements if `x`, `y`, `omega` all have the same shape. (In that case, people can simply write the formula for the multiplier using element-wise operations, without having to broadcast.) I think that we adopted a similar strategy for the laser profiles.